### PR TITLE
ADCS Driver Update - HAVT

### PR DESCRIPTION
**Pull request by shihaocao**
This PR adds HAVT read and write capability to the ADCS Driver.
It also fixes some constants.

Here is the sister PR on ADCSSoftware:
https://github.com/pathfinder-for-autonomous-navigation/ADCSSoftware/pull/24

### CODE REVIEW

File: src/adcs_test.cpp [hunk 1]
```diff
@@ -128,6 +131,16 @@ bool test_set_imu_mode(){
...
 }
 
+bool test_set_havt(){
+
+    //replace with MAX devices
+    std::bitset<32> cmd(0x0F0F0F0F);
+
+    adcs.set_havt(cmd);
+
+    return true;
```
[the remark this answers is not in the document]
Comment (shihaocao) [reply src/adcs_test.cpp]: That's correct, none of the set() commands are actually rigorously tested. I'm verifying that they work by looking at the debug console for the ADCS Computer.

For what it's worth, I have manually checked that it is indeed being set correctly.

File: lib/Drivers/ADCS.cpp [hunk 2]
```diff
@@ -250,21 +286,43 @@ void ADCS::get_ssa_vector(std::array<float, 3>* ssa_sun_vec) {
...
 
-void ADCS::update_hat() {}
\ No newline at end of file
+void ADCS::get_havt(std::bitset<MAX_DEVICES>* havt_table){
+    unsigned char temp[4];
+    std::memset(temp, 0, sizeof(temp));
+
+    #ifdef UNIT_TEST
+    for(int i = 0;i<4;i++){
+        temp[i] = 255;
```
[the remark this answers is not in the document]
Comment (shihaocao) [reply lib/Drivers/ADCS.cpp]: I think I'm going to choose to set all ones to be ALL functional. That way, I can directly read the series of is_functional() into the bitset.

File: lib/Drivers/ADCS.cpp [hunk 3]
```diff
@@ -124,46 +133,69 @@ void ADCS::set_ssa_voltage_filter(const float voltage_filter) {
...
     float_decomp(kd, cmd);
     i2c_write_to_subaddr(IMU_GYR_TEMP_KD,cmd,4);
 }
+
 void ADCS::set_imu_gyr_temp_desired(const float desired){
     unsigned char cmd = uc(desired,imu::min_eq_temp,imu::max_eq_temp);
     i2c_write_to_subaddr(IMU_GYR_TEMP_DESIRED,cmd);
 }
 
+void ADCS::set_havt(const std::bitset<MAX_DEVICES>& havt_table){
```
[the remark this answers is not in the document]
Comment (shihaocao) [reply lib/Drivers/ADCS.cpp]: Yes the controller will do the mapping. I clarified in ADCS.hpp

>/**
     * @brief Get the adcs havt table bitset
     * 
     * Access the state of the 1st device using havt_table.test(0);
     * 2nd device is at havt_table.test(1);
     * 32nd device is at havt_table.test(31);
     * 
     * If a bit of the bitset is high, that device's is_functional() returned true.
     * 
     * @param havt_table Pointer to the bitset that will be read into
     */

File: lib/Drivers/ADCS.hpp [hunk 4]
```diff
@@ -10,12 +10,19 @@
...
+
 namespace Devices {
 
 class ADCS : public I2CDevice {
    public:
     static constexpr unsigned int ADDRESS = 0x4E;
     static constexpr unsigned int WHO_AM_I_EXPECTED = 0x0F;
+
+    //TODO MOVE TO COMMON SOFTWARE? USED FOR BITSET CAPACITY
+    static constexpr unsigned int MAX_DEVICES = 32;
```
[the remark this answers is not in the document]
Comment (shihaocao) [reply lib/Drivers/ADCS.hpp]: Just pushed a PR on common software for that

File: lib/Drivers/ADCS.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (shihaocao) [reply lib/Drivers/ADCS.cpp]: Clarified in ADCS.hpp

File: lib/Drivers/ADCS.hpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (tanishqaggarwal) [reply lib/Drivers/ADCS.hpp]: Do you want to remove it from here then?

[the remark this answers is not in the document]
Comment (shihaocao) [reply lib/Drivers/ADCS.hpp]: Right, i'll go ahead and use the common software const now

**Comment by shihaocao** [comment]
@tanishqaggarwal Sorry for the big delay in implementing that simple change, but I merely swapped out all constants with the CommonSoftware havt constant. Should be pretty straight forward.

**Pull request merged**